docs/release-notes/breaking-change-policy.asciidoc

```diff
@@ -0,0 +1,26 @@
+[[breaking-changes-policy]]
+=== Breaking changes policy
+
+
+The {java-client} source code is generated from a https://github.com/elastic/elasticsearch-specification[formal specification of the Elasticsearch API]. This API specification is large, and although it is tested against hundreds of Elasticsearch test files, it may have discrepancies with the actual API that result in issues in the {java-client}.
+
+Fixing these discrepancies in the API specification result in code changes in the {java-client}, and some of these changes can require code updates in your applications.
+
+This section explains how these breaking changes are considered for inclusion in {java-client} releases.
+
+[discrete]
+==== Breaking changes in patch releases
+
+Some issues in the API specification are properties that have an incorrect type, such as a `long` that should be a `string`, or a required property that is actually optional. These issues can cause the {java-client} to not work properly or even throw exceptions.
+
+When these specification issues are found and fixed, they may require code updates in applications using the {java-client}. These are considered acceptable, _even in patch releases_ (e.g. 7.17.0 -> 7.17.1), as they fix issues that would otherwise make the corresponding APIs unusable.
```

Comment:
```suggestion
When a specification issue is discovered and resolved, it may require code update in applications using the {java-client}. These breaking changes, _even in patch releases_ (e.g. 7.17.0 -> 7.17.1), are considered acceptable as it introduces stability and otherwise the corresponding API might unusable.
```